### PR TITLE
Nftmarket v2 implementation

### PR DESCRIPTION
**Pull request by jununifi**

### CODE REVIEW

File: proto/nftmarket/nftmarket.proto [hunk 1]
```diff
@@ -43,18 +43,14 @@ message ListedNft {
...
-  ];
-  uint64                    bid_active_rank = 7;
-  google.protobuf.Timestamp started_at      = 8
+  NftIdentifier             nft_id               = 1 [(gogoproto.nullable) = false];
+  string                    owner                = 2;
+  ListingType               listing_type         = 3;
+  ListingState              state                = 4;
+  string                    bid_token            = 5;
+  uint64                    minimum_deposit_rate = 6;
+  bool                      automatic_overdraft  = 7;
```
Comment (jununifi) [inline proto/nftmarket/nftmarket.proto:52]: Can you share the meaning of automatic overdraft @mkXultra ?

File: proto/nftmarket/nftmarket.proto [hunk 2]
```diff
@@ -43,18 +43,14 @@ message ListedNft {
...
-    (gogoproto.nullable)   = false
-  ];
-  uint64                    bid_active_rank = 7;
-  google.protobuf.Timestamp started_at      = 8
+  NftIdentifier             nft_id               = 1 [(gogoproto.nullable) = false];
+  string                    owner                = 2;
+  ListingType               listing_type         = 3;
+  ListingState              state                = 4;
+  string                    bid_token            = 5;
+  uint64                    minimum_deposit_rate = 6;
```
Comment (jununifi) [inline proto/nftmarket/nftmarket.proto:51]: What's the unit of deposit rate?

Comment (mkXultra) [reply proto/nftmarket/nftmarket.proto:51]: This makes little sense.
Always lister would enter the lowest deposit interest rate the lowest.
According to Kimura, advanced users need it.

If a bid is submitted with a bid deposit below the minimum interest rate, the bid will be rejected

File: proto/nftmarket/nftmarket.proto (hunk 1 above)
Comment (mkXultra) [reply proto/nftmarket/nftmarket.proto:52]: this case
https://github.com/UnUniFi/chain/blob/f116b8eb5e51caf7d930d8942b27c617419ddc1a/docs/architecture/adr-nft-market/05.UX.md?plain=1#L24

File: proto/nftmarket/nftmarket.proto [hunk 3]
```diff
@@ -69,16 +65,18 @@ message NftListing {
...
+  NftIdentifier            nft_id         = 1 [(gogoproto.nullable) = false];
+  string                   bidder         = 2;
+  cosmos.base.v1beta1.Coin bid_amount     = 3 [(gogoproto.nullable) = false];
+  string                   deposit_amount = 4 [
     (gogoproto.moretags)   = "yaml:\"paid_amount\"",
     (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
     (gogoproto.nullable)   = false
   ];
-  google.protobuf.Timestamp bid_time = 6
+  google.protobuf.Timestamp bidding_period       = 5 [(gogoproto.nullable) = false];
```
[the remark this answers is not in the document]
Comment (mkXultra) [reply proto/nftmarket/nftmarket.proto:76]: I will ask Mr. Kimura.

File: proto/nftmarket/nftmarket.proto [hunk 4]
```diff
@@ -69,16 +65,18 @@ message NftListing {
...
+  uint64                    deposit_lending_rate = 6 [(gogoproto.nullable) = false];
```
[the remark this answers is not in the document]
Comment (mkXultra) [reply proto/nftmarket/nftmarket.proto:77]: it is persent.
range is 1 to 100.

File: proto/nftmarket/nftmarket.proto (hunk 2 above)
Comment (jununifi) [reply proto/nftmarket/nftmarket.proto:51]: I mean if deposit rate is in percentage value or if the unit is basis point (1/10000)

File: proto/nftmarket/nftmarket.proto (hunk 1 above)
Comment (jununifi) [reply proto/nftmarket/nftmarket.proto:52]: Deposit is done from multiple bids, if it's returned to the bidder who borrowed, what would be the result of this action? Borrow move move to other bids?

File: proto/nftmarket/nftmarket.proto (hunk 3 above)
[the remark this answers is not in the document]
Comment (jununifi) [reply proto/nftmarket/nftmarket.proto:76]: Okay

File: proto/nftmarket/nftmarket.proto (hunk 4 above)
[the remark this answers is not in the document]
Comment (jununifi) [reply proto/nftmarket/nftmarket.proto:77]: Okay, thanks for confirmation

File: proto/nftmarket/nftmarket.proto [hunk 5]
```diff
@@ -13,12 +13,10 @@ enum ListingType {
   LATE_SHIPPING            = 2;
 }
 enum ListingState {
-  LISTING          = 0;
-  BIDDING          = 1;
-  SELLING_DECISION = 2;
-  LIQUIDATION      = 3;
```
Comment (jununifi) [inline proto/nftmarket/nftmarket.proto:19]: Selling decision and liquidation status removed - can you confirm the change? @mkXultra

File: proto/nftmarket/tx.proto [hunk 6]
```diff
@@ -12,17 +12,13 @@ service Msg {
...
   rpc PlaceBid(MsgPlaceBid) returns (MsgPlaceBidResponse);
   rpc CancelBid(MsgCancelBid) returns (MsgCancelBidResponse);
   rpc SellingDecision(MsgSellingDecision) returns (MsgSellingDecisionResponse);
   rpc EndNftListing(MsgEndNftListing) returns (MsgEndNftListingResponse);
   rpc PayFullBid(MsgPayFullBid) returns (MsgPayFullBidResponse);
   rpc Borrow(MsgBorrow) returns (MsgBorrowResponse);
   rpc Repay(MsgRepay) returns (MsgRepayResponse);
-  rpc MintStableCoin(MsgMintStableCoin) returns (MsgMintStableCoinResponse);
-  rpc BurnStableCoin(MsgBurnStableCoin) returns (MsgBurnStableCoinResponse);
-  rpc Liquidate(MsgLiquidate) returns (MsgLiquidateResponse);
```
Comment (jununifi) [inline proto/nftmarket/tx.proto:25]: Removed ExpandListingPeriod, MintStableCoin, BurnStableCoin, Liquidate messages - can you confirm the changes ? @mkXultra

File: proto/nftmarket/tx.proto [hunk 7]
```diff
@@ -5,175 +5,134 @@ import "gogoproto/gogo.proto";
...
-    (gogoproto.moretags) = "yaml:\"sender\"",
+    (gogoproto.moretags)   = "yaml:\"sender\"",
     (gogoproto.customtype) = "github.com/UnUniFi/chain/types.StringAccAddress",
-    (gogoproto.nullable) = false
+    (gogoproto.nullable)   = false
   ];
-  NftIdentifier nft_id = 2 [ (gogoproto.nullable) = false ];
-  ListingType listing_type = 3;
-  string bid_token = 4;
-  string min_bid = 5 [
```
Comment (Senna46) [inline proto/nftmarket/tx.proto:50]: Why does u delete `min_bid`?
The lister cannot decide this, is that a problem?

**Pull request merged**